### PR TITLE
Disable Docker's health check until we officially support it

### PR DESCRIPTION
**Pull request by yguo0905**
Ref: https://github.com/kubernetes/kubernetes/issues/50703

Tested locally.

Without this PR:

```
CONTAINER ID        IMAGE                                                  COMMAND                  CREATED             STATUS                     PORTS               NAMES
afdd796ddddc        gcr.io/ygg-gke-dev/nginx-healthcheck                   "nginx -g 'daemon ..."   5 minutes ago       Up 5 minutes (healthy)                       k8s_test-container_test-pod_default_8a1ad225-82bf-11e7-becb-480fcf514648_0
```

With this PR:
```
e3fb2437555f        gcr.io/ygg-gke-dev/nginx-healthcheck                   "nginx -g 'daemon ..."   10 seconds ago      Up 9 seconds                            k8s_test-container_test-pod_default_de82e83c-82c0-11e7-b3fc-480fcf514648_0
```

**Release note**:

```
Disable Docker's health check until we officially support it.
```

/assign @yujuhong 
/assign @dchen1107

### CODE REVIEW

File: pkg/kubelet/dockershim/docker_container.go [hunk 1]
```diff
@@ -131,6 +131,10 @@ func (ds *dockerService) CreateContainer(podSandboxID string, config *runtimeapi
 			OpenStdin: config.Stdin,
 			StdinOnce: config.StdinOnce,
 			Tty:       config.Tty,
+			// Disable Docker's health check until we officially support it.
+			Healthcheck: &dockercontainer.HealthConfig{
```
Comment (dchen1107) [inline pkg/kubelet/dockershim/docker_container.go]: Is this compatible with docker 1.10 and 1.11?

Comment (yguo0905) [inline pkg/kubelet/dockershim/docker_container.go]: This is the compatibility between new docker client API and old docker daemon, which should not be an issue?

Comment (yguo0905) [inline pkg/kubelet/dockershim/docker_container.go]: But on the other hand, to cherry-pick this into 1.7 and 1.6, we need https://github.com/kubernetes/kubernetes/pull/48308..

Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container.go]: :-(

File: pkg/kubelet/dockershim/docker_container.go [hunk 2]
```diff
@@ -131,6 +131,10 @@ func (ds *dockerService) CreateContainer(podSandboxID string, config *runtimeapi
 			OpenStdin: config.Stdin,
 			StdinOnce: config.StdinOnce,
 			Tty:       config.Tty,
+			// Disable Docker's health check until we officially support it.
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container.go]: Add the issue number in the comment #25829

File: pkg/kubelet/dockershim/docker_container.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container.go]: Unfortunately, the docker-related code changed *a lot* in the past 3 releases. Updating the docker client for each release will be a completely different PR....
I think it might be too intrusive. Let's think it through first...

File: pkg/kubelet/dockershim/docker_container.go (hunk 2 above)
Comment (yguo0905) [inline pkg/kubelet/dockershim/docker_container.go]: Done. Added a link.

**Comment by yujuhong** [comment]
LGTM, but leaving @dchen1107 a chance to object

/retest

File: pkg/kubelet/dockershim/docker_container.go [hunk 3]
```diff
@@ -131,6 +131,11 @@ func (ds *dockerService) CreateContainer(podSandboxID string, config *runtimeapi
 			OpenStdin: config.Stdin,
 			StdinOnce: config.StdinOnce,
 			Tty:       config.Tty,
+			// Disable Docker's health check until we officially support it
+			// (https://github.com/kubernetes/kubernetes/issues/25829).
+			Healthcheck: &dockercontainer.HealthConfig{
+				Test: []string{"NONE"},
+			},
```
Comment (asacbp) [inline pkg/kubelet/dockershim/docker_container.go]: fwiw, this causes lots of warnings in docker log like: 

Dec 11 08:50:29 bpdehamkubn01 dockerd[14450]: time="2017-12-11T08:50:29.993217562+01:00" level=warning msg="Unknown healthcheck type 'NONE' (expected 'CMD') in container XXXXX"

is this really the cleanest way to tell docker to not use the healthcheck?

Comment (dims) [reply pkg/kubelet/dockershim/docker_container.go]: @asacbp - that's what they are recommending, https://github.com/docker/engine-api/blob/master/types/container/config.go#L16 So maybe worth asking on the docker github issues?

Comment (yujuhong) [reply pkg/kubelet/dockershim/docker_container.go]: This has just been fixed in docker, but hasn't been included in any release yet. 
https://github.com/moby/moby/pull/35533
